v4/core/request_builder.go

```diff
@@ -98,13 +99,78 @@ func (requestBuilder *RequestBuilder) ConstructHTTPURL(serviceURL string, pathSe
 		}
 
 		if pathParameters != nil && i < len(pathParameters) {
+			if pathParameters[i] == "" {
+				return requestBuilder, fmt.Errorf(ERRORMSG_PATH_PARAM_EMPTY, fmt.Sprintf("[%d]", i))
```

Comment:
In the old function, there's no way for us to know the name of the path parameter so I went ahead and just included its index in the "pathParameters" array within the error string.  It's certainly not ideal, but the index is at least related to the order in which the path param was defined within the operation.

Reply:
> I'm trying to understand your comment above.. Did you mean to have
> 
> ```
> 				return requestBuilder, fmt.Errorf(ERRORMSG_PATH_PARAM_EMPTY, fmt.Sprintf("[%d]", pathParameters[i]))
> ```
> 
> here instead of just returning the index ?

No, in the new function, we actually have access to the name of the path param that might have a value of "", so the error message would be something like: `path parameter 'resource_id' is empty`
However, in the old function, we were not previously checking for a "" value, so when I added that check I needed to somehow identify which path parameter has the "" value.   The only thing I had access to was the index into the pathParameters array, so the error message would look something like:  `path parameter '[1]' is empty`
It's not great, but it is something :)